core/src/main/java/software/amazon/awssdk/runtime/StandardMemberCopier.java

```diff
@@ -74,6 +74,19 @@ public static ByteBuffer copy(ByteBuffer bb) {
         if (bb == null) {
             return null;
         }
-        return bb.duplicate();
+
+        ByteBuffer copy = bb.isDirect() ? ByteBuffer.allocateDirect(bb.capacity()) : ByteBuffer.allocate(bb.capacity());
+
+        // Copy content
+        ByteBuffer toCopy = bb.asReadOnlyBuffer();
+        toCopy.rewind();
```

Comment:
why do we want to rewind? We shouldn't care about data that's not in this byte buffers view right? Also can you use the implementation in BinaryUtils?